90zfsbootmenu/zfsbootmenu-lib.sh

```diff
@@ -2120,6 +2121,65 @@ zbmcmdline() {
   [ -f "${BASE}/zbm.cmdline" ] && echo | cat "${BASE}/zbm.cmdline" -
 }
 
+# arg1: pid
+# prints: child pid
+# returns: 0 if a child pid was found, 1 if there are no children
+
+find_child_pid() {
+  local pid child
+
+  pid="${1}"
+  if [ -z "${pid}" ]; then
+    zdebug "empty pid"
+    return 1
+  fi
+
+  if [ -e "/proc/${pid}/task/${pid}/children" ] ; then
```

Comment:
Any idea how widely supported this is?

Reply:
I'll have to see when this was added to the `/proc` vfs - I image it's been there for quite a few major kernel versions though.

Reply:
Looks like it's been there since at least 2012 - https://lwn.net/Articles/475688/. I think we can safely assume it's always available.

Reply:
Are you really prepared to abandon support for anything before the latest 2.6 kernels? That probably makes us incompatible with Slackware.

Reply:
Don't forget CentOS 5.